packages/flutter/lib/src/widgets/table.dart

```diff
@@ -394,48 +399,51 @@ class _TableElement extends RenderObjectElement {
       }
     }
     final Iterator<_TableElementRow> oldUnkeyedRows = _children.where((_TableElementRow row) => row.key == null).iterator;
-    final List<_TableElementRow> newChildren = <_TableElementRow>[];
-    final Set<List<Element>> taken = <List<Element>>{};
+
+    final List<Element> allOldChildren = <Element>[];
+    final List<Widget> allNewChildren = <Widget>[];
+
     for (final TableRow row in newWidget.children) {
       List<Element> oldChildren;
       if (row.key != null && oldKeyedRows.containsKey(row.key)) {
         oldChildren = oldKeyedRows[row.key]!;
-        taken.add(oldChildren);
-      } else if (row.key == null && oldUnkeyedRows.moveNext()) {
+        oldKeyedRows.remove(row.key);
+      } else if (oldUnkeyedRows.moveNext()) {
         oldChildren = oldUnkeyedRows.current.children;
       } else {
         oldChildren = const <Element>[];
       }
+
+      allOldChildren.addAll(oldChildren);
+      allNewChildren.addAll(row.children!);
+    }
+    while (oldUnkeyedRows.moveNext())
+      allOldChildren.addAll(oldUnkeyedRows.current.children);
+    oldKeyedRows.values.forEach(allOldChildren.addAll);
+
+    final List<Element> updatedChildren = updateChildren(allOldChildren, allNewChildren, forgottenChildren: _forgottenChildren);
```

Comment:
If we want to use the slot index, we must update all the children (call `updateChildren()`) one-time, otherwise, the calculation of the slot is wrong.

This is why make the change here.